### PR TITLE
Dg gawb 3400

### PR DESCRIPTION
**Pull request by dags**
### Description

Upload feature had issues while selecting un-expanded folders due to the lazy load approach used to improve performance. 

https://broadinstitute.atlassian.net/browse/BTRX-41

### Checklist

- [X ] **Submitter**: Include the JIRA issue number in the PR description
- [X ] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [X] **Submitter**: Ensure that you have added or updated tests
- [ ] **Submitter**: If you're adding/switching libraries or otherwise changing the design, update the [design documentation](https://broadinstitute.atlassian.net/wiki/pages/viewpage.action?pageId=189431877).
- [ ] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * Add notes on what you've tested
* Review cycle:
  * Team may comment on PR at will
  * **Reviewer assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to reviewer** for further feedback
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Squash commits and merge to develop
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Move the JIRA issue to QA once this checklist is completed

### CODE REVIEW

File: electron_app/UploadManager.js [hunk 1]
```diff
@@ -11,7 +12,19 @@ const uploadManager = (bucketName, fileList = [], access_token, win) => {
...
+    if (file.preserveStructure === true) {
+      if (process.platform === 'win32') {
+        uri = uri + 'Uploads' + file.path.slice(file.path.indexOf(':') + 1).replace(/\\/g , '/');
+      } else {
+        uri = uri + 'Uploads' + file.path;
+      }
+    } else {
+      uri = uri + 'Uploads/' + file.name;
+    }
+
```
Comment (davidangb) [inline electron_app/UploadManager.js:27]: Between this change and #73 and others, might want to have a centralized path/filesystem utility that knows how to interpret Windows vs. other paths - or, look for a library that handles it.

Not necessary to merge this PR - this is an idea for the future!

**Review by davidangb — COMMENTED** [review]
One idea inline, not required for this PR. Code looks good; I have not tested.

From a process perspective, since I'm just starting to review your PRs ... has this already been reviewed/tested internally? I want to understand the level of detail I should put into my review. Thanks!

File: electron_app/UploadManager.js (hunk 1 above)
Comment (dags) [reply electron_app/UploadManager.js:27]: makes sense, we'll take into account from now on. Thanks.

**Comment by dags** [comment]
Regarding the process: every PR is reviewed and tested internally. I think you should not worry about testing, unless something is obviously wrong.

**Review by davidangb — APPROVED** [review]
:+1: approved, thanks!

**Pull request merged**